core/shared/src/main/scala/org/http4s/Uri.scala

```diff
@@ -929,8 +929,10 @@ object Uri extends UriPlatform {
     val target = (base, reference) match {
       case (_, Uri(Some(_), _, _, _, _)) => reference
       case (Uri(s, _, _, _, _), Uri(_, a @ Some(_), p, q, f)) => Uri(s, a, p, q, f)
-      case (Uri(s, a, p, q, _), Uri(_, _, pa, Query.empty, f)) if pa.isEmpty => Uri(s, a, p, q, f)
-      case (Uri(s, a, p, _, _), Uri(_, _, pa, q, f)) if pa.isEmpty => Uri(s, a, p, q, f)
+      case (Uri(s, a, p, q, _), Uri(_, _, pa, Query.empty, f)) if pa.isEmpty && !pa.absolute =>
```

Comment:
_a skippable minor grumpy nitpick._: does `pa.isEmpty && !pa.absolute` merely correspond to `pa == Path.empty`?

Reply:
Hmm, that's a good question. It also opens the question that should `pa.isEmpty` be equivalent to `pa == Path.empty`? Is `/` empty or not? 😵

Reply:
It's interesting, maybe, it's kinda side effect that `Path.empty` equals `/`.